### DISCUSSION TITLE
Flag for `zfs send` exists in docs but not in latest release

### DISCUSSION THREAD

**Discussion by erkannt** [openzfs/zfs#14361] [Q&A] [answered]
## Context

- The `-X` flag for `zfs-send` is supposed to make it possible to exclude datasets from recursive (`-R`) send operations.
- Flag and behaviour is specified in [current (2023-01-03) documentation](https://openzfs.github.io/openzfs-docs/man/8/zfs-send.8.html#X).
- Flag is not available in zfs version 2.1.7:
  ```
  # zfs version
  zfs-2.1.7-1
  zfs-kmod-2.1.7-1
  # zfs send  -vpn -R -X raw-data tank
  invalid option 'X'
  ```
- The `-X` flag and associated code where merged to `master` in March 2022.  
  https://github.com/openzfs/zfs/pull/13158

## Questions

- Is it intentional that this feature hasn't made it into the releases since the merge into `master`?
- Should the documentation site perhaps indicate which version it covers?
  - Version selection on the docs site?
  - Only publish docs for latest release?
  - Note in the docs which version introduced a feature?

**Comment by gmelikov** [comment, marked as answer]
https://openzfs.github.io/openzfs-docs/man/ tracks `master` branch now, we track version difference task in this issue https://github.com/openzfs/openzfs-docs/issues/2 , but it's not in active development now. Contributions are welcome, of course.

> Is it intentional that this feature hasn't made it into the releases since the merge into master?

Yes, stable releases include safe easily-backporting changes or bug fixes (primarily). But anyone is welcome to propose inclusion of new changes into stable branch, if one is ready to backport it and test thoroughly. It's critical to have `stable` branch stable.

> Should the documentation site perhaps indicate which version it covers? 

Yes, it would be great not to only describe version, but to have a mechanism to read man pages for exact version. Maybe it's a good idea to mention somehow on that pages about version. At least `Edit on GitHub` link is already leads to master branch.
